interfaces/apparmor/backend.go

```diff
@@ -285,6 +286,24 @@ func profileGlobs(snapName string) []string {
 	return []string{interfaces.SecurityTagGlob(snapName), nsProfile(snapName)}
 }
 
+// Determine if a profile filename is removable during core refresh/rollback.
+// On systems with unified cache directory, don't remove the snap profiles,
+// only system and snap-confine profiles. snap-confine profiles are like the
+// following:
+// - usr.lib.snapd.snap-confine.real
+// - usr.lib.snapd.snap-confine
+// - snap.core.NNNN.usr.lib.snapd.snap-confine
+// - var.lib.snapd.snap.core.NNNN.usr.lib.snapd.snap-confine
+// - snap-confine.core.NNNN
+// TODO: also the "snapd" snap here soon
+func profileIsRemovableOnCoreSetup(fn string) bool {
```

Comment:
Could you add a comment that explains why a profile is or is not removable in the general sense. I understand that this is so that snapd boots but I don't know if I'm right and I suspect most people will quickly forget the rationale unless it's written down.

Reply:
This part of why I didn't have it broken out into a separate function before because all this is related to the comment in Setup() below, where this is called. I'll (essentially) just dupe the comment from below.

Reply:
Aha, I see now. Thank you!